Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoadManager.cpp

```diff
@@ -379,19 +379,22 @@ void SpeculativeLoadManager::registerLoad(GlobalFrameID frameID, const ResourceR
         ASSERT(!m_pendingFrameLoads.contains(frameID));
 
         // Start tracking loads in this frame.
-        auto pendingFrameLoad = PendingFrameLoad::create(protectedStorage(), resourceKey, [this, frameID] {
-            bool wasRemoved = m_pendingFrameLoads.remove(frameID);
+        auto pendingFrameLoad = PendingFrameLoad::create(protectedStorage(), resourceKey, [weakThis = WeakPtr { *this }, frameID] {
+            if (!weakThis)
+                return;
+            CheckedPtr checkedThis = weakThis.get();
```

Comment:
We normally convert the WeakPtr to a non-weak one *and then* null check that new pointer, rather than null checking the WeakPtr first. It doesn't make a difference in single-threaded code but it is still good practice.

Reply:
My bad, fixed now.